lib/ClangImporter/ImportDecl.cpp

```diff
@@ -2394,7 +2395,8 @@ namespace {
               llvm_unreachable("unknown decl kind");
 
             IdStream << "__Unnamed_" << kind << "_";
-            if (field->isAnonymousStructOrUnion()) {
+            if (field->isAnonymousStructOrUnion() ||
+                field->getDeclName().isEmpty()) {
```

Comment:
I don't think this change is actually necessary—we only get here for a `TagDecl` (i.e. a struct or union decl) with no name. When you use the Microsoft extension, the `TagDecl` does have a name.

(But an `assert(!field->getDeclName().isEmpty() && "MS anonymous struct extension?")` in the `else` branch would be a sensible precaution.)